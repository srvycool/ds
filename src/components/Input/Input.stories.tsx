import React, { useState } from 'react';
import { Input } from './Input';

export default {
  title: 'Input',
  component: Input,
};

export const Default: React.FC = () => {
  const [value, setValue] = useState(`Aute duis enim culpa nostrud`);

  return (
    <>
      <Input
        value={value}
        onChange={(event) => setValue(event.target.value)}
        marginBottom="l"
      />
      <Input
        label="With label"
        value={value}
        onChange={(event) => setValue(event.target.value)}
      />
    </>
  );
};
