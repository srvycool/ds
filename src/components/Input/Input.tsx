import React from 'react';
import { Box } from '../Box/Box';
import { Typography } from '../Typography/Typography';
import styled, { StyledComponentProps } from 'styled-components';
import { MarginBottomProps } from 'styled-system';
import { Theme } from '../../theme';

export type InputProps = StyledComponentProps<'input', Theme, {}, never> &
  MarginBottomProps<Theme> & {
    label?: string;
  };

const StyledInputBox = styled(Box)`
  transition: 100ms border-color ease-in-out;

  &:focus-within {
    border-color: ${(props) => props.theme.colors.primary500};
  }
`;

const StyledInput = styled.input`
  ${(props) => props.theme.typography.body1};
  color: ${(props) => props.theme.colors.grey900};
  border: 0;
  width: 100%;
  padding: 0;

  &:focus {
    outline: none;
  }
`;

export const Input: React.FC<InputProps> = ({
  label,
  marginBottom,
  ...props
}) => {
  return (
    <Box as="label" display="block" marginBottom={marginBottom}>
      {label && (
        <Typography
          display="block"
          variant="caption"
          color="grey800"
          marginBottom="xs"
        >
          {label}
        </Typography>
      )}
      <StyledInputBox
        as="span"
        display="block"
        boxShadow="primary"
        border="primary"
        padding="s"
        width="100%"
        borderRadius="primary"
        position="relative"
      >
        <StyledInput {...props} />
      </StyledInputBox>
    </Box>
  );
};
