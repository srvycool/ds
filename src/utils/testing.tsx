import React from 'react';
import {
  render as tlrender,
  RenderOptions,
  RenderResult,
} from '@testing-library/react';
import { ThemeProvider } from '../components/ThemeProvider/ThemeProvider';

export function render(
  ui: React.ReactElement,
  options?: Omit<RenderOptions, 'queries'>
): RenderResult {
  return tlrender(<ThemeProvider>{ui}</ThemeProvider>, options);
}
