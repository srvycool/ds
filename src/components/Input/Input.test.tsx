import React from 'react';
import { render } from '../../utils/testing';
import { Input } from './Input';

describe('Input', () => {
  describe('without label', () => {
    it('should match snapshot', () => {
      const { container } = render(
        <Input
          value="Non culpa ullamco est laboris pariatur."
          onChange={() => null}
        />
      );

      expect(container).toMatchSnapshot();
    });
  });

  describe('without label', () => {
    it('should match snapshot', () => {
      const { container } = render(
        <Input
          label="With Label"
          value="Non culpa ullamco est laboris pariatur."
          onChange={() => null}
        />
      );

      expect(container).toMatchSnapshot();
    });
  });
});
