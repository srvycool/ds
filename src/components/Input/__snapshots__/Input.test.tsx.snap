// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input without label should match snapshot 1`] = `
.c0 {
  display: block;
  box-sizing: border-box;
}

.c1 {
  display: block;
  box-shadow: 0px 1px 2px rgba(0,0,0,0.06),0px 1px 3px rgba(0,0,0,0.1);
  border: 2px solid #E6DFE1;
  padding: 12px;
  width: 100%;
  border-radius: 8px;
  position: relative;
  box-sizing: border-box;
  -webkit-transition: 100ms border-color ease-in-out;
  transition: 100ms border-color ease-in-out;
}

.c1:focus-within {
  border-color: #DA4167;
}

.c2 {
  font-size: 16px;
  line-height: 1.2;
  font-family: Montserrat;
  color: #493C3F;
  border: 0;
  width: 100%;
  padding: 0;
}

.c2:focus {
  outline: none;
}

<div>
  <label
    class="c0"
    display="block"
  >
    <span
      class="c1"
      display="block"
      width="100%"
    >
      <input
        class="c2"
        value="Non culpa ullamco est laboris pariatur."
      />
    </span>
  </label>
</div>
`;

exports[`Input without label should match snapshot 2`] = `
.c0 {
  display: block;
  box-sizing: border-box;
}

.c1 {
  margin: 0;
  color: #7D666C;
  margin-bottom: 8px;
  display: block;
  font-size: 12px;
  line-height: 1.2;
  font-family: Montserrat;
  text-transform: uppercase;
}

.c2 {
  display: block;
  box-shadow: 0px 1px 2px rgba(0,0,0,0.06),0px 1px 3px rgba(0,0,0,0.1);
  border: 2px solid #E6DFE1;
  padding: 12px;
  width: 100%;
  border-radius: 8px;
  position: relative;
  box-sizing: border-box;
  -webkit-transition: 100ms border-color ease-in-out;
  transition: 100ms border-color ease-in-out;
}

.c2:focus-within {
  border-color: #DA4167;
}

.c3 {
  font-size: 16px;
  line-height: 1.2;
  font-family: Montserrat;
  color: #493C3F;
  border: 0;
  width: 100%;
  padding: 0;
}

.c3:focus {
  outline: none;
}

<div>
  <label
    class="c0"
    display="block"
  >
    <span
      class="c1"
      color="grey800"
      display="block"
    >
      With Label
    </span>
    <span
      class="c2"
      display="block"
      width="100%"
    >
      <input
        class="c3"
        value="Non culpa ullamco est laboris pariatur."
      />
    </span>
  </label>
</div>
`;
